### PR TITLE
framesh: init at 0.5.0-beta.20

### PR DESCRIPTION
**Pull request by 0xnook**
###### Description of changes
Added a package for [frame](https://github.com/floating/frame) as requested by #168384

Went with `framesh` as opposed to `frame` to avoid collision with existing package in nixpkgs.

Decided to package the appimage as opposed to building from source since releases are provided this way, have tested this release for a while with no problems, and I personally find it cumbersome to deal with npm derivations.
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2205.section.md [hunk 1]
```diff
@@ -941,6 +941,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 - The Nextcloud module now supports to create a Mysql database automatically
   with `services.nextcloud.database.createLocally` enabled.
 
+- The `framesh` new package has been added, which lets users interact with Web3 dapss.
```
Comment (K900) [inline nixos/doc/manual/release-notes/rl-2205.section.md:944]: s/dapss/dapps/?

File: pkgs/applications/blockchains/framesh/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    install -m 444 -D ${appimageContents}/frame.desktop $out/share/applications/frame.desktop
+    install -m 444 -D ${appimageContents}/frame.png \
+    $out/share/icons/hicolor/512x512/apps/frame.png
+    substituteInPlace $out/share/applications/frame.desktop \
+    --replace 'Exec=AppRun' 'Exec=${pname}'
+  '';
+
+  meta = {
+    mainProgram = "framesh";
+    description = "A native web3 interface, that lets you interact with dapps";
```
Comment (K900) [inline pkgs/applications/blockchains/framesh/default.nix:30]: Extra comma.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -14678,4 +14678,10 @@
     github = "snpschaaf";
     githubId = 105843013;
   };
+  nook = {
```
Comment (K900) [inline maintainers/maintainer-list.nix:14681]: This is usually alphabetized.

File: nixos/doc/manual/from_md/release-notes/rl-2205.section.xml [hunk 4]
```diff
@@ -2688,6 +2688,12 @@ cp /var/lib/redis/dump.rdb &quot;/var/lib/redis-mastodon/dump.rdb&quot;
           enabled.
         </para>
       </listitem>
+      <listitem>
```
Comment (K900) [inline nixos/doc/manual/from_md/release-notes/rl-2205.section.xml:2691]: Should probably go into the 22.11 release notes now, if at all.

File: pkgs/applications/blockchains/framesh/default.nix (hunk 2 above)
Comment (0xnook) [reply pkgs/applications/blockchains/framesh/default.nix]: removed comma

File: nixos/doc/manual/release-notes/rl-2205.section.md (hunk 1 above)
Comment (0xnook) [reply nixos/doc/manual/release-notes/rl-2205.section.md]: fixed typo

File: nixos/doc/manual/from_md/release-notes/rl-2205.section.xml (hunk 4 above)
Comment (0xnook) [reply nixos/doc/manual/from_md/release-notes/rl-2205.section.xml]: moved to 22.11 release notes

File: nixos/doc/manual/from_md/release-notes/rl-2211.section.xml [hunk 5]
```diff
@@ -207,9 +210,41 @@
...
     </itemizedlist>
+    <blockquote>
+      <blockquote>
+        <blockquote>
+          <blockquote>
+            <blockquote>
+              <blockquote>
+                <blockquote>
+                  <para>
+                    bd33059f025 (framesh: move release notes from 2205
```
Comment (K900) [inline nixos/doc/manual/from_md/release-notes/rl-2211.section.xml:239]: Definitely a merge conflict.

File: nixos/doc/manual/release-notes/rl-2211.section.md [hunk 6]
```diff
@@ -69,12 +69,18 @@ Use `configure.packages` instead.
 
 ## Other Notable Changes {#sec-release-22.11-notable-changes}
 
+<<<<<<< HEAD
```
Comment (K900) [inline nixos/doc/manual/release-notes/rl-2211.section.md:72]: Fix it here and regenerate the docbook files.

File: pkgs/applications/blockchains/framesh/default.nix [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  };
+in
+appimageTools.wrapType2 {
+  inherit name src;
+
+  extraInstallCommands = ''
+    ln -s $out/bin/${name} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/frame.desktop $out/share/applications/frame.desktop
+    install -m 444 -D ${appimageContents}/frame.png \
+    $out/share/icons/hicolor/512x512/apps/frame.png
```
Comment (K900) [inline pkgs/applications/blockchains/framesh/default.nix:23]: Minor nitpick: maybe indent those a bit more so it's easier to read? Something like

```bash
install -m 444 -D ${appimageContents}/frame.png \
    $out/share/icons/hicolor/512x512/apps/frame.png
```

File: nixos/doc/manual/release-notes/rl-2211.section.md [hunk 8]
```diff
@@ -69,12 +69,18 @@ Use `configure.packages` instead.
...
 - The `xplr` package has been updated from 0.18.0 to 0.19.0, which brings some breaking changes. See the [upstream release notes](https://github.com/sayanarijit/xplr/releases/tag/v0.19.0) for more details.
 
 - A new module was added for the Saleae Logic device family, providing the options `hardware.saleae-logic.enable` and `hardware.saleae-logic.package`.
 
 - Matrix Synapse now requires entries in the `state_group_edges` table to be unique, in order to prevent accidentally introducing duplicate information (for example, because a database backup was restored multiple times). If your Synapse database already has duplicate rows in this table, this could fail with an error and require manual remediation.
 
 - memtest86+ was updated from 5.00-coreboot-002 to 6.00-beta2. It is now the upstream version from https://www.memtest.org/, as coreboot's fork is no longer available.
+=======
+- A new module was added for the Saleae Logic device family, providing the options `hardware.saleae-logic.enable` and `hardware.saleae-logic.package`.
```
[the remark this answers is not in the document]
Comment (0xnook) [reply nixos/doc/manual/release-notes/rl-2211.section.md:81]: Sorry about this! Indeed a mismerge, will fix

File: pkgs/applications/blockchains/framesh/default.nix (hunk 7 above)
Comment (0xnook) [reply pkgs/applications/blockchains/framesh/default.nix]: added indentation for the multiline command

File: pkgs/applications/blockchains/framesh/default.nix [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+in
+appimageTools.wrapType2 {
+  inherit name src;
+
+  extraInstallCommands = ''
+    ln -s $out/bin/${name} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/frame.desktop $out/share/applications/frame.desktop
+    install -m 444 -D ${appimageContents}/frame.png \
+      $out/share/icons/hicolor/512x512/apps/frame.png
+      substituteInPlace $out/share/applications/frame.desktop \
```
Comment (K900) [inline pkgs/applications/blockchains/framesh/default.nix:24]: Too much indentation here. That's why it matters :)

File: nixos/doc/manual/release-notes/rl-2211.section.md (hunk 6 above)
Comment (0xnook) [reply nixos/doc/manual/release-notes/rl-2211.section.md:72]: yes, on it

File: pkgs/applications/blockchains/framesh/default.nix [hunk 10]
```diff
@@ -0,0 +1,38 @@
+{ lib, fetchurl, appimageTools, ... }:
```
Comment (K900) [inline pkgs/applications/blockchains/framesh/default.nix:1]: Another nitpick: the `...` is unnecessary.

File: pkgs/applications/blockchains/framesh/default.nix [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    url = "https://github.com/floating/frame/releases/download/v${version}/Frame-${version}.AppImage";
+    sha256 = "sha256-/mmVnUnJVfoKfXMN2zPlo+lsUninCNuENmVA2x8dDEY=";
+  };
+
+  appimageContents = appimageTools.extractType2 {
+    inherit name src;
+  };
+in
+appimageTools.wrapType2 {
+  inherit name src;
```
Comment (K900) [inline pkgs/applications/blockchains/framesh/default.nix:17]: Also better to set `pname` and `version` separately.

File: pkgs/applications/blockchains/framesh/default.nix [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+  pname = "framesh";
+  version = "0.5.0-beta.19";
+  name = "${pname}-${version}";
+  src = fetchurl {
+    url = "https://github.com/floating/frame/releases/download/v${version}/Frame-${version}.AppImage";
+    sha256 = "sha256-/mmVnUnJVfoKfXMN2zPlo+lsUninCNuENmVA2x8dDEY=";
+  };
+
+  appimageContents = appimageTools.extractType2 {
+    inherit name src;
```
Comment (K900) [inline pkgs/applications/blockchains/framesh/default.nix:13]: I don't think `name` is necessary here at all.

Comment (0xnook) [reply pkgs/applications/blockchains/framesh/default.nix]: ``appimageTools.extractType2 {} `` requires either name or pname and version.

Comment (K900) [reply pkgs/applications/blockchains/framesh/default.nix]: All right. Not too familiar with AppImage stuff myself.

File: nixos/doc/manual/release-notes/rl-2211.section.md [hunk 13]
```diff
@@ -73,6 +73,8 @@ Use `configure.packages` instead.
 
 - A new module was added for the Saleae Logic device family, providing the options `hardware.saleae-logic.enable` and `hardware.saleae-logic.package`.
 
+- The `framesh` new package has been added, which lets users interact with Web3 dapps.
```
Comment (K900) [inline nixos/doc/manual/release-notes/rl-2211.section.md:76]: Another super tiny nit: "new package has been added" sounds weird. Maybe drop the "new"?

File: pkgs/applications/blockchains/framesh/default.nix [hunk 14]
```diff
@@ -0,0 +1,38 @@
+{ lib, fetchurl, appimageTools }:
+
+let
+  pname = "framesh";
+  version = "0.5.0-beta.19";
+  name = "${pname}-${version}";
```
Comment (K900) [inline pkgs/applications/blockchains/framesh/default.nix:6]: And this is unused now.

Comment (0xnook) [reply pkgs/applications/blockchains/framesh/default.nix:6]: it is used for symlinking the binary name, but can get rid of the variable declaration

**Review by K900 — APPROVED** [review]
Very very minor nitpicks left, the package builds and runs, I don't know if it actually works as intended because I don't have (or want to deal with) crypto. Otherwise LGTM. Also please squash all the fixup commits.

File: pkgs/applications/blockchains/framesh/default.nix (hunk 14 above)
Comment (K900) [reply pkgs/applications/blockchains/framesh/default.nix:6]: Maybe inline it then.

**Comment by 0xnook** [comment]
Thanks for the in depth review @K900

File: pkgs/applications/blockchains/framesh/default.nix [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+    ln -s $out/bin/${pname}-${version} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/frame.desktop $out/share/applications/frame.desktop
+    install -m 444 -D ${appimageContents}/frame.png \
+      $out/share/icons/hicolor/512x512/apps/frame.png
+    substituteInPlace $out/share/applications/frame.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
+  '';
+
+  meta = {
+    mainProgram = "framesh";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/framesh/default.nix:28]: Doesn't that default already to this?

**Review by SuperSandro2000 — COMMENTED** [review]
Please squash the commits according to the contributing guide.

File: pkgs/applications/blockchains/framesh/default.nix [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+    description = "A native web3 interface that lets you interact with dapps";
```
[the remark this answers is not in the document]
Comment (0xnook) [reply pkgs/applications/blockchains/framesh/default.nix:29]: According to [nixos manual](https://nixos.org/manual/nixpkgs/stable/#var-meta-description), this should be a one line description.

So how about: "Native web3 interface that lets you interact with dapps via web3 protocols like ethereum" and keep the long description

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/blockchains/framesh/default.nix:29]: > According to [nixos manual](https://nixos.org/manual/nixpkgs/stable/#var-meta-description), this should be a one line description.

It is one line. GitHub just wraps it around.



> So how about: "Native web3 interface that lets you interact with dapps via web3 protocols like ethereum" and keep the long description

The long one is a bit short for a long description and does not contain to many additional information. Usually long descriptions go over ~5 or more lines.

[the remark this answers is not in the document]
Comment (0xnook) [reply pkgs/applications/blockchains/framesh/default.nix:29]: Perhaps one line in the manual is a bit vague and open to interpretation, but ill go with your suggested changes

File: nixos/doc/manual/release-notes/rl-2211.section.md [hunk 17]
```diff
@@ -119,6 +119,8 @@ Use `configure.packages` instead.
 
 - The Redis module now disables RDB persistence when `services.redis.servers.<name>.save = []` instead of using the Redis default.
 
+- The `framesh` package has been added, which lets users interact with Web3 dapps.
+
```
[the remark this answers is not in the document]
Comment (0xnook) [reply nixos/doc/manual/release-notes/rl-2211.section.md]: Removed release notes and squashed commits

File: pkgs/applications/blockchains/framesh/default.nix (hunk 15 above)
Comment (0xnook) [reply pkgs/applications/blockchains/framesh/default.nix]: Indeed, removed

**Comment by 0xnook** [comment]
Thanks for your review and comments @SuperSandro2000 :smiley: I believe I am done implementing your revisions

**Pull request merged**